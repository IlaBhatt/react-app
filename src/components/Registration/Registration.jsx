// author @POD-1
import { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import { makeStyles } from "@material-ui/core";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { Card, CardContent } from "@mui/material";
import "../../styles/Registration.css";
import { keyCloakRegisterUser, getToken } from "../../UserService";

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",

    "& .MuiTextField-root": {
      margin: theme.spacing(2),
      width: "300px",
    },
    "& .MuiButtonBase-root": {
      margin: theme.spacing(2),
    },
  },
}));

const Registration = () => {
  useEffect(() => {
    getToken();
  }, []);

  const navigate = useNavigate();

  // console.log(localStorage.getItem("bearer-token"));
  const classes = useStyles();
  // create state variables for each input
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [telephone, setTelephone] = useState("");
  const [error, setError] = useState(false);
  const [userExists, setUserExists] = useState(false);
  // const [bgColor, setbgColor] = useState("#FFFFFF");

  const handleSubmit = (e) => {
    e.preventDefault();

    // password checker
    const validatePass =
      /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=[^ ]{8,15}$)/;

    if (!validatePass.test(password)) {
      setError(true);
      // error msg toastify
      return;
    } else {
      setError(false);
    }

    const raw = JSON.stringify({
      enabled: true,
      email,
      firstName,
      lastName,
      credentials: [
        {
          type: "password",
          value: password,
          temporary: false,
        },
      ],
      attributes: {
        telephone,
      },
    });

    keyCloakRegisterUser(raw)
      .then((response) => response.text())
      .then((result) => {
        if (result === "") {
          setUserExists(false);
          // saving the username password to local storage for email verification
          localStorage.setItem("email", email);
          localStorage.setItem("firstName", firstName);
          navigate("./email-verification-page");
        } else {
          setUserExists(true);
        }
      });
  };

  return (
    <>
      <div className="registration-bg-image">
        <div className="registration-cardContainer">
          <Card
            variant="outlined"
            raised = {true}
            className="mdc-card mdc-card--outlined registration-my-card"
            style={{
              minWidth: "sm",
              backgroundColor: "#FFFFFF",
              borderRadius: "25px",
              paddingTop: 30,
            }}
          >
            <h4>Sign Up</h4>
            <CardContent>
              <form className={classes.root} onSubmit={handleSubmit}>
                <TextField
                  id="filled-basic"
                  inputProps={{ "data-testid": "fname" }}
                  label="First Name"
                  variant="filled"
                  required
                  value={firstName}
                  onChange={(e) => setFirstName(e.target.value)}
                />
                <TextField
                  id="filled-basic"
                  inputProps={{ "data-testid": "lname" }}
                  label="Last Name"
                  variant="filled"
                  required
                  value={lastName}
                  onChange={(e) => setLastName(e.target.value)}
                  style={{ borderRadius: "25px", backgroundColor: "#FFFFFF" }}
                />
                <TextField
                  id="filled-basic"
                  inputProps={{ "data-testid": "email" }}
                  label="Email"
                  variant="filled"
                  type="email"
                  required
                  error={userExists}
                  helperText={userExists ? "Username Already Exists" : ""}
                  value={email}
                  onChange={(e) => {
                    setUserExists(false);
                    setEmail(e.target.value);
                  }}
                />

                <TextField
                  id="filled-basic"
                  inputProps={{ "data-testid": "password" }}
                  label="Password"
                  variant="filled"
                  type="password"
                  required
                  value={password}
                  onChange={(e) => {
                    setError(false);
                    setPassword(e.target.value);
                  }}
                  error={error}
                  helperText={error ? "Invalid Password" : ""}
                  // style={{
                  //   borderRadius: "25px",
                  //   backgroundColor: "FFFFFF",
                  // }}
                />
                <TextField
                  id="filled-basic"
                  inputProps={{ "data-testid": "telephone" }}
                  label="Telephone Number"
                  variant="filled"
                  type="tel"
                  value={telephone}
                  onChange={(e) => setTelephone(e.target.value)}
                  // style={{ borderRadius: "25px", backgroundColor: "#FFFFFF" }}
                />
                <div>
                  <Button
                    inputProps={{ "data-testid": "cancel" }}
                    variant="contained"
                    // onClick={handleClose}
                    style={{ color: "#FFFFFF", backgroundColor: "#24a0ed" }}
                  >
                    Cancel
                  </Button>
                  <Button
                    inputProps={{ "data-testid": "signup" }}
                    color="primary"
                    onSubmit={handleSubmit}
                    type="submit"
                    variant="contained"
                    style={{ color: "#FFFFFF", backgroundColor: "#24a0ed" }}
                  >
                    Sign up
                  </Button>
                </div>
              </form>
            </CardContent>
          </Card>
        </div>
      </div>
    </>
  );
};

export default Registration;
