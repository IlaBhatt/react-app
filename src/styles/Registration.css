 /* author @POD-1 */
.registration-my-card {
  /* margin-top: 30px;
  
  /* margin: 5rem 45rem; */
  /* min-width: 40%;
  max-width: 70%;*/
  margin-bottom: 10%;
  margin-top: 5%;
}
/* body {
  background-image: url("https://mediaim.expedia.com/localexpert/529178/b973c47b-98cc-4c92-8f2d-faae9144b4d1.jpg");
} */

.registration-title {
  width: fit-content;
  height: fit-content;
  font-size: 30px;
  font-weight: bold;
  margin-left: 5%;
  margin-top: 1%;
}
.registration-cardContainer {
  margin-right: 11%;
  justify-content: flex-end;
  display: flex;
}
.registration-bg-image {
  background-image: url("https://mediaim.expedia.com/localexpert/529178/b973c47b-98cc-4c92-8f2d-faae9144b4d1.jpg");
  width: 100%;
  height: 100%;
}
/* .card {
  display: flex;
  min-width: 20%;
   flex-direction: column;
  justify-content: space-between; 
  align-items: center;
} */
