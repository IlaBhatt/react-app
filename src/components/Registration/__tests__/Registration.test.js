// author @POD-1
import { shallow, render } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Registration from "../Registration";
import { TextField, Button } from "@material-ui/core";
import { Card, CardContent } from "@mui/material";
import { BrowserRouter as Router } from "react-router-dom";
import { findRenderedComponentWithType } from "react-dom/test-utils";



describe("Registration", function () {
  let registration;
  beforeEach(() => {
    registration = shallow(
      <Router>
        <Registration />
      </Router>
    );
  });

    it("renders registration page correctly", () => {
      const tree = renderer.create(registration).toJSON();
      expect(tree).toMatchSnapshot();
    });
  // it("renders a sign up button", () => {
  //   render(<Button />);

  //   expect().toBe("Sign Up");
  // });
  // const mockFn = jest.fn();
  // it('should call mock function when button is clicked', () => {
  //     const tree = render(
  //       <Button id= "signup" onSubmit={mockFn} />
  //     );
  //     tree.simulate('click');
  //     expect(mockFn).toHaveBeenCalled();
  //   });

  it("renders the form currently", () => {
    const component = shallow(<form />);
    expect(component.length).toBe(1);
  });

  it("renders the card currently", () => {
    const component = shallow(<Card />);
    expect(component.length).toBe(1);
  });
  it("renders the card content currently", () => {
    const component = shallow(<CardContent />);
    expect(component.length).toBe(1);
  });

  it("renders first name text field input", () => {
    const fname = render(<TextField id="fname" />);
    // const firstName = getByTestId("fname");
    expect(fname.length).toBe(1);
  });

  it("renders last name text field input", () => {
    const lname = render(<TextField id="lname" />);
    expect(lname.length).toBe(1);
  });

  it("renders email text field input", () => {
    const email = render(<TextField id="email" />);
    expect(email.length).toBe(1);
  });

  it("renders password text field input", () => {
    const password = render(<TextField id="password" />);
    expect(password.length).toBe(1);
  });

  it("renders telephone text field input", () => {
    const telephone = render(<TextField id="telephone" />);
    expect(telephone.length).toBe(1);
  });

  describe("Button", function () {
    let mountedButton;

    beforeEach(() => {
      mountedButton = render(<Button />);
    });

    it("renders without crashing", () => {
      mountedButton = render(<Button />);
      expect(mountedButton.length).toBe(1);
    });

    // it('renders a button', () => {
    //     const bt = mountedButton.find(<Button/>);
    //     expect(bt.length).toBe(2);
    // });
    it("matches snapshot", () => {
      const tree = renderer.create(<Button id="signup"></Button>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("renders the sign up button", () => {
      const btn = render(<Button id="signup" />);
      expect(btn.length).toBe(1);
    });

    it("renders the cancel button", () => {
      const cancelbtn = render(<Button id="cancel" />);
      expect(cancelbtn.length).toBe(1);
    });
  });

  // it("Sign up submit button should be able to redirect to the home page", () => {
  //   shallow(/Login)
  // })
});
