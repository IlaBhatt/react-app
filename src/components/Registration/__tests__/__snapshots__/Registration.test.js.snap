// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Button matches snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text"
  disabled={false}
  id="signup"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Registration renders registration page correctly 1`] = `
<div
  className="registration-bg-image"
>
  <div
    className="registration-cardContainer"
  >
    <div
      className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root mdc-card mdc-card--outlined registration-my-card css-1v5z4dq-MuiPaper-root-MuiCard-root"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": "25px",
          "minWidth": "sm",
          "paddingTop": 30,
        }
      }
    >
      <h4>
        Sign Up
      </h4>
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <form
          className="makeStyles-root-1"
          onSubmit={[Function]}
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-required Mui-required"
              data-shrink={false}
              htmlFor="filled-basic"
              id="filled-basic-label"
            >
              First Name
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input"
                data-testid="fname"
                disabled={false}
                id="filled-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": "25px",
              }
            }
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-required Mui-required"
              data-shrink={false}
              htmlFor="filled-basic"
              id="filled-basic-label"
            >
              Last Name
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input"
                data-testid="lname"
                disabled={false}
                id="filled-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-required Mui-required"
              data-shrink={false}
              htmlFor="filled-basic"
              id="filled-basic-label"
            >
              Email
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input"
                data-testid="email"
                disabled={false}
                id="filled-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="email"
                value=""
              />
            </div>
            
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-required Mui-required"
              data-shrink={false}
              htmlFor="filled-basic"
              id="filled-basic-label"
            >
              Password
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input"
                data-testid="password"
                disabled={false}
                id="filled-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
                value=""
              />
            </div>
            
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink={false}
              htmlFor="filled-basic"
              id="filled-basic-label"
            >
              Telephone Number
            </label>
            <div
              className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input"
                data-testid="telephone"
                disabled={false}
                id="filled-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="tel"
                value=""
              />
            </div>
          </div>
          <div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained"
              disabled={false}
              inputProps={
                Object {
                  "data-testid": "cancel",
                }
              }
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "backgroundColor": "#24a0ed",
                  "color": "#FFFFFF",
                }
              }
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Cancel
              </span>
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              inputProps={
                Object {
                  "data-testid": "signup",
                }
              }
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onSubmit={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "backgroundColor": "#24a0ed",
                  "color": "#FFFFFF",
                }
              }
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Sign up
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
